### PR TITLE
feat: new sleep timer (integrated into the player UI)

### LINKED ISSUE #5146: Move sleep timer to the player.
### Describe your suggested feature

Right now, to enable/disable or change the sleep timer's duration, we
need to restart the app, and we also need to go to the settings to do it.

I propose that it should be given as a option in the video player
making it easy to enable/disable it.

### Other details

AntennaPod has a similar feature.

### Acknowledgements

- [X] I have searched the existing issues and this is a new ticket, **NOT** a duplicate or related to another open issue.
- [X] I have written a short but informative title.
- [X] I will fill out all of the requested information in this form.

### PR DESCRIPTION
**Pull request by Bnyro**
closes #5146

### CODE REVIEW

File: app/src/main/java/com/github/libretube/ui/sheets/SleepTimerSheet.kt [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+        updateTimeLeftText()
+
+        binding.startSleepTimer.setOnClickListener {
+            val time = binding.timeInput.text.toString().toLongOrNull()
+
+            if (time == null) {
+                Toast.makeText(context, R.string.invalid_input, Toast.LENGTH_SHORT).show()
+                return@setOnClickListener
+            }
+            SleepTimer.setup(context as BaseActivity, time)
```
Comment (Isira-Seneviratne) [inline app/src/main/java/com/github/libretube/ui/sheets/SleepTimerSheet.kt:45]: This casting doesn't seem necessary.

File: app/src/main/java/com/github/libretube/ui/tools/SleepTimer.kt [hunk 2]
```diff
@@ -6,52 +6,67 @@ import android.os.Looper
...
+            timeLeftMillis -= TIMER_DELAY
+            if (timeLeftMillis == 0L) showTimerEndedSnackBar(context)
+        }
+    }
+
+    /**
+     * Disable the scheduled sleep timer
+     */
+    fun disableSleepTimer() {
+        if (this::timer.isInitialized) timer.cancel()
```
Comment (Isira-Seneviratne) [inline app/src/main/java/com/github/libretube/ui/tools/SleepTimer.kt:42]: The timer could be a nullable variable instead, that way this couls be simplified to timer?.cancel().

**Comment by Isira-Seneviratne** [comment]
This looks good to me overall, haven't tested it out though.

**Comment by tusharhero** [comment]
Thank you considering this feature!

**Pull request merged**